src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs

```diff
@@ -572,6 +589,19 @@ private IList<SelectorModel> CreateSelectors(IList<object> attributes)
             return selectorModels;
         }
 
+        private static bool InRouteProviders(List<IRouteTemplateProvider> routeProviders, object attribute)
+        {
+            foreach (var rp in routeProviders)
+            {
+                if (ReferenceEquals(rp, attribute))
+                {
+                    return true;
+                }
```

Comment:
Sadly there's nothing on List that takes a comparer except for binary search, and this isn't sorted. Linear search still seems best because we're talking about attributes on a method `0 < N < ~10`